util/error.ts

```diff
@@ -4,10 +4,3 @@ export function ErrorCtor<Name extends string>(name: Name) {
     override readonly name = name;
   };
 }
-
-export function throwIfError<T>(value: T): Exclude<T, Error> {
```

Comment:
It's used in 3 tests... an `instanceof` check is slightly larger, but it's clear what's happening in place (without following the import). In the case of examples: we want it to be **extremely clear** to users what's going on. My thinking is that `throwIfError` comes with an unworthy trade off: the cognitive load of bringing `throwIfError` into scope (and figuring out what it does) isn't worth the size savings. That being said, I agree: avoids the local binding on which to perform the `instanceof` check... are we (am I) overthinking this? Should we revert?

Reply:
It's used in 3 tests... and pretty much every single example. While we don't use it much in capi (since we tend to return results), I think it's a very useful utility for users and makes the examples a bit cleaner.

In terms of cognitive overhead, once one knows what it does, that's not a problem. Whereas with the expanded version, one has to read it to make sure it's the boilerplate and not doing something weird.

Reply:
That's a good point! I agree. Reverting now.